index.html

```diff
@@ -2069,194 +2074,239 @@
+                            <li>If <var>option</var>.<a data-lt=
+                            "PaymentShippingOption.selected">selected</a> is
+                            true, then set <var>selectedShippingOption</var> to
+                            <var>option</var>.<a data-lt=
+                            "PaymentShippingOption.id">id</a>.
+                            </li>
+                          </ol>
                         </li>
-                        <li>Add <var>serializedData</var> to
-                        <var>serializedModifierData</var>.
+                      </ol>
+                    </li>
+                    <li>If the <a>modifiers</a> member of <var>details</var> is
+                    present, then:
+                      <ol>
+                        <li>Let <var>modifiers</var> be the sequence
+                        <var>details</var>.<a>modifiers</a>.
                         </li>
-                        <li>Remove the <a data-lt=
-                        "PaymentDetailsModifier.data">data</a> member of <var>
-                          modifier</var>, if it is present.
+                        <li>Let <var>serializedModifierData</var> be an empty
+                        list.
+                        </li>
+                        <li>For each <a>PaymentDetailsModifier</a>
+                        <var>modifier</var> in <var>modifiers</var>:
+                          <ol data-link-for="PaymentDetailsModifier">
+                            <li>If the <a>total</a> member of
+                            <var>modifier</var> is present and
+                            <var>member</var>.<a>total</a>.<a data-lt=
+                            "PaymentItem.amount">amount</a>.<a data-lt=
+                            "PaymentCurrencyAmount.value">value</a> is not a
+                            <a>valid decimal monetary value</a>, then <a>abort
+                            the update</a> with a <a>TypeError</a>.
+                            </li>
+                            <li>If the <a>additionalDisplayItems</a> member of
+                            <var>modifier</var> is present, then for each
+                            <a>PaymentItem</a> <var>item</var> in
+                            <var>modifier</var>.<a>additionalDisplayItems</a>:
+                              <ol>
+                                <li>Let <var>amountValue</var> be
+                                <var>item</var>.<a data-lt=
+                                "PaymentItem.amount">amount</a>.<a data-lt=
+                                "PaymentCurrencyAmount.value">value</a>.
+                                </li>
+                                <li>If <var>amountValue</var> is not a <a>valid
+                                decimal monetary value</a>, then <a>abort the
+                                update</a> with a <a>TypeError</a>.
+                                </li>
+                                <li>If the first character of
```

Comment:
It should be OK to have negative line items in `additionalLineItems`. This can be "Debit card discount," for example. Let's reomve this `<li>...</li>`.